frontend/public/components/deploy-image.tsx

```diff
@@ -0,0 +1,439 @@
+    {(portsArr.length === 1 || (portsArr.length > 1 && i === portsArr.length - 1)) && <React.Fragment> </React.Fragment>}
+    {(portsArr.length > 1 && i !== portsArr.length - 1) && <React.Fragment>, </React.Fragment>}
+  </span>)}
+  will be load balanced by Service <strong>{name || '<name>'}</strong>.
+  <div>Other containers can access this service through the hostname <strong>{name || '<name>'}</strong>.</div>
+</React.Fragment>;
+
+export class DeployImage extends React.Component<DeployImageProps, DeployImageState> {
+  constructor (props) {
+    super(props);
+
+    this.state = {
+      namespace: '',
+      imageName: '',
+      loading: false,
+      inProgress: false,
+      name: '',
+    };
+
+    this.onNamespaceChange = this.onNamespaceChange.bind(this);
+    this.onImageNameChange = this.onImageNameChange.bind(this);
+    this.onKeyPress = this.onKeyPress.bind(this);
+    this.onNameChange = this.onNameChange.bind(this);
+    this.search = this.search.bind(this);
+    this.save = this.save.bind(this);
+  }
+
+  onNamespaceChange = (namespace: string) => {
+    this.setState({namespace: namespace});
+  };
+
+  onImageNameChange: React.ReactEventHandler<HTMLInputElement> = event => {
+    this.setState({imageName: event.currentTarget.value});
+  }
+
+  onKeyPress = event => {
+    if (this.state.imageName !== '' && event.key === 'Enter') {
+      this.search(event);
+    }
+  }
+
+  onNameChange: React.ReactEventHandler<HTMLInputElement> = event => {
+    this.setState({name: event.currentTarget.value});
+  };
+
+  search = event => {
+    event.preventDefault();
+
+    const { namespace, imageName } = this.state;
+    const importImage = {
+      kind: 'ImageStreamImport',
+      apiVersion: 'image.openshift.io/v1',
+      metadata: {
+        name: imageName,
+        namespace: namespace
+      },
+      spec: {
+        import: false,
+        images: [{
+          from: {
+            kind: 'DockerImage',
+            name: imageName
+          }
+        }]
+      },
+      status: {}
+    };
+
+    this.setState({
+      loading: true,
+      isi: null,
+      name: '',
+      error: null,
+      searchError: null,
+    });
+
+    k8sCreate(ImageStreamImportsModel, importImage)
+      .then((imageStreamImport) => {
+        const name = _.get(imageStreamImport, 'spec.images[0].from.name');
+        const status = _.get(imageStreamImport, 'status.images[0].status');
+        if (status.status === 'Success') {
+          this.setState({
+            loading: false,
+            isi: {
+              name,
+              image: _.get(imageStreamImport, 'status.images[0].image'),
+              tag: _.get(imageStreamImport, 'status.images[0].tag'),
+              status
+            },
+            name,
+          });
+        } else {
+          this.setState({
+            loading: false,
+            searchError: _.upperFirst(status.message),
```

Comment:
I'd leave off `upperFirst`. Typically we don't do that in the converged console, and it's not right for some messages